model/map/src/main/java/org/keycloak/models/map/authorization/MapPolicyStore.java

```diff
@@ -111,8 +111,8 @@ public void delete(String id) {
     public Policy findById(String id, String resourceServerId) {
         LOG.tracef("findById(%s, %s)%s", id, resourceServerId, getShortStackTrace());
 
-        return tx.read(forResourceServer(resourceServerId)
-                .compare(SearchableFields.ID, Operator.EQ, id))
+        return tx.read(withCriteria(forResourceServer(resourceServerId)
+                .compare(SearchableFields.ID, Operator.EQ, id)))
                 .findFirst()
```

Comment:
@hmlnarik I am just looking into this piece of code and it seems we should add the pagination to places like this. I think, we should go through the source code and optimize with pagination where possible. WDYT? And also should we do it as part of this PR or create and enhancement Jira in the cleanup epic? I would probably prefer the second option as this PR is already quite big.

Reply:
+1 for follow-up JIRA. It seems to be nice to have rather than must have, right?